pkg/compact/compact.go

```diff
@@ -994,9 +993,9 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 	}
 
 	var toCompact []*metadata.Meta
-	tracing.DoInSpanWithErr(ctx, "compaction_planning", func(ctx context.Context) error {
-		toCompact, err = planner.Plan(ctx, cg.metasByMinTime)
-		return err
+	err = tracing.DoInSpanWithErr(ctx, "compaction_planning", func(ctx context.Context) (e error) {
+		toCompact, e = planner.Plan(ctx, cg.metasByMinTime)
```

Comment:
I guess we declare `e` before. Using `err` here should be fine if I am not mistaken...